### PR TITLE
Feature/chapter16

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
*[AP - Chapter 16]  Taking pictures with intents - https://androidtraining.atlassian.net/browse/AT-103

## Description:
Chapter 16 created

## :camera_flash: UI preview
ezgif com-video-to-gif(17)

### CODE REVIEW

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 1]
```diff
@@ -83,43 +114,195 @@ class CrimeFragment : Fragment(R.layout.fragment_crime), DatePickerFragment.Call
...
+
+  private fun buttonsListener() {
+    CrimeFragmentHelper().pickerButtons()
+
+    reportButton.setOnClickListener {
+      Intent(Intent.ACTION_SEND).apply {
+        type = TYPE_INTENT
+        putExtra(Intent.EXTRA_TEXT, CrimeFragmentHelper().getCrimeReport())
+        putExtra(Intent.EXTRA_SUBJECT, getString(R.string.crime_report_subject))
+      }.also { intent ->
```
Comment (f7deleon) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt]: why you are using alson in this section?

File: app/src/main/res/layout/fragment_crime.xml [hunk 2]
```diff
@@ -9,19 +9,59 @@
...
+                android:scaleType="centerInside"
+                />
+
+            <ImageButton
+                android:id="@+id/crimeCamera"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:src="@android:drawable/ic_menu_camera"
+                />
+        </LinearLayout>
```
Comment (f7deleon) [inline app/src/main/res/layout/fragment_crime.xml:41]: nl

File: app/src/main/res/layout/fragment_crime.xml [hunk 3]
```diff
@@ -9,19 +9,59 @@
...
+                />
+
+            <EditText
+                android:id="@+id/crimeTitle"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:hint="@string/title_hint"
+                />
+        </LinearLayout>
+    </LinearLayout>
```
Comment (f7deleon) [inline app/src/main/res/layout/fragment_crime.xml:64]: nls

**Review by f7deleon — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -20,6 +20,25 @@
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <provider
+            android:name="androidx.core.content.FileProvider"
+            android:authorities="com.bignerdranch.android.criminalintent.fileprovider"
+            android:exported="false"
+            android:grantUriPermissions="true"
+            >
+            <meta-data
```
Comment (mirland) [inline app/src/main/AndroidManifest.xml:30]: nl

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 5]
```diff
@@ -83,43 +114,195 @@ class CrimeFragment : Fragment(R.layout.fragment_crime), DatePickerFragment.Call
...
+
+        val contactUri: Uri? = data.data
+        val queryFields = arrayOf(
+          ContactsContract.CommonDataKinds.Phone.NUMBER,
+          ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME
+        )
+        val cursor = requireActivity().contentResolver.query(
+          contactUri!!, queryFields, null, null, null
+        )
+        cursor?.use {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt]: repository or something like that

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 6]
```diff
@@ -83,43 +114,195 @@ class CrimeFragment : Fragment(R.layout.fragment_crime), DatePickerFragment.Call
...
+        }
+      }
+      resultCode == REQUEST_PHOTO -> {
+        requireActivity().revokeUriPermission(photoUri, Intent.FLAG_GRANT_WRITE_URI_PERMISSION)
+        CrimeFragmentHelper().updatePhotoView()
+      }
+    }
+  }
+
+  inner class CrimeFragmentHelper {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt]: why are you  using another class?

File: app/src/main/java/com/bignerdranch/android/criminalintent/database/CrimeDatabase.kt [hunk 7]
```diff
@@ -3,10 +3,18 @@ package com.bignerdranch.android.criminalintent.database
...
 import com.bignerdranch.android.criminalintent.models.Crime
 
-@Database(entities = [Crime::class], version = 1)
+@Database(entities = [Crime::class], version = 2)
 @TypeConverters(CrimeTypeConverters::class)
 abstract class CrimeDatabase : RoomDatabase() {
   abstract fun crimeDao(): CrimeDao
 }
+
+val migration_1_2 = object : Migration(1, 2) {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/database/CrimeDatabase.kt]: it should be private... I'd add a database povider

File: app/src/main/java/com/bignerdranch/android/criminalintent/models/Crime.kt [hunk 8]
```diff
@@ -10,5 +10,9 @@ data class Crime(
     @PrimaryKey val id: UUID = UUID.randomUUID(),
     var title: String = "",
     var date: Date = Date(),
-    var isSolved: Boolean = false
-)
+    var isSolved: Boolean = false,
+    var suspect: String = ""
+) {
+  val photoFileName
+    get() = "IMG_$id.jpg"
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/models/Crime.kt:20]: constrant

Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/models/Crime.kt:20]: you don't need the get() property

File: app/src/main/res/xml/files.xml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<paths>
+    <files-path
+        name="crimePhotos"
+        path="."
+        />
+</paths>
```
Comment (mirland) [inline app/src/main/res/xml/files.xml]: nl

**Review by FerrariBruno — APPROVED** [review]

**Pull request merged**